src/components/Data/Documents/DocumentListItem.vue

```diff
@@ -147,7 +147,10 @@ export default {
     },
     editDocument() {
       if (this.canEdit) {
-        this.$emit('edit', this.document.id)
+        this.$router.push({
```

Comment:
I don't agree with this. Such a "leaf" component shouldn't take the responsibility of pushing the whole app to a new route (thus be coupled to the route naming). Parent component should be in charge of deciding what happens when the user asks to edit a document.

Reply:
Why it shouldn't?  I thought the component approach was made for this, to have more "independent" components.  

Here I don't see why the parent should be in charge of this. Everywhere this component is rendered it has the same meaning and if the user click on update then we should redirect him to the update page.

Otherwise we have to pass method and event alongside the component chain which is a very tedious task.

IMHO the parent component should not have the responsibility of the navigation either, like applying some rights or idk. This should only be in the router, what do you think?

Reply:
Exactly, components should be independent and the idea is to minimize coupling, especially with the route names and params, because they're totally application-specific.
Leaf components should be kept as light as possible and IMHO the good practice is always to ask yourself if the same component would work in another application.
The argument that we're not using this component in another application is not a valid one to justify coupling.

Root-ish components are usually more coupled to the application-specific logic and it is more acceptable to make them aware of routes, for example, especially because, most of the time, we want to tend to displace this logic toward the roots, so that we can play more easily with leaf component and keep the UI lean.

I'll give you an example.
Imagine that tomorrow we decide to move the document edition to a modal or an inline editor right in the same page instead of doing it in another page. It would be something the Page component would be aware, of course, and if we keep the leaf component away from this concern, we'll have to modify only the parent component.

Reply:
I agree that passing the method alongside the component tree is tedious, this is why I always try to keep things simple by making leaf components just `$emit` and the parent just listen to `@event="$router.push()`

I agree that, when it comes to situations where we have several degrees of component nesting, it is tedious to forward the events from children to parents. I think the good compromise is "try not to couple leaf components while trying not to have too much depth in the component tree".

Reply:
But I see your point: you are disturbed by having to pass such a tiny event from the Item :point_right: List :point_right: Page and I agree it's overkill to pass an item through 3 components just to go to another page. So, in this case, I think yours is the best solution.